💭 멘토님 리뷰

eunjju2 opened the pull request:
## 🛠️ 서비스 요약

- 사용자와 사업자가 하나의 플랫폼에서 연결되는 스터디룸 예약 서비스
- 사용자는 위치 및 장소 검색을 통해 스터디룸을 찾고 예약 및 결제를 할 수 있으며, 사업자는 본인의 스터디룸과 예약자를 관리할 수 있습니다.

## 📝 요구 사항과 구현 내용

1. 로그인 및 회원가입, 사업장 등록 및 스터디룸 상세 보기 - `최성령`
2. 지도 및 장소 검색, 예약 및 결제, 1:1 문의 - `이은수`
3. 사용자 & 사업자 회원 관리, 사업장 및 예약 관리, 실시간 알림 - `조현진`

## 📝 작업 현황

- 프론트측 페이지별 마크업은 마무리된 상태입니다. api 연결하면서 자잘한 부분만 수정이 이루어지면 될 것 같습니다.
- 백엔드쪽 서버 배포가 돼서 현재 가능한 api를 연결해서 테스트해보고 있습니다. 다만 완벽히 배포된 것은 아니어서 테스트에 시간이 조금 소요되고 있습니다.
    
    지금 테스트해보고 있는 페이지는 
    
    - 사업자 / 사용자 회원가입 페이지
    - 사업자 / 사용자 로그인 페이지
    - 사업자 / 사용자 마이페이지(로그아웃 포함)
    - 메인페이지
    - 사업장 상세페이지
    
    정도입니다. 중간발표때까지 해당 페이지들은 최대한 구현을 끝낼 예정입니다. 
    

## 💬 리뷰 시에 참고사항

- 로컬 구동 시 `npm install` 실행하셔야 합니다.
- .env 파일에 각 키값을 넣어야 하는데 DM으로 요청해주시면 바로 알려드리겠습니다.
- 지도 이용 시 브라우저에서 현재 위치 수집에 관한 권한이 동의되어야 합니다.
- 각 페이지 경로는 Rounter.tsx 참고하시면 됩니다.

## 💬 작업내역에 대한 설명 & 궁금한 점

- typings 폴더에는 공통으로 사용하는 타입을 정리해두고, constants 폴더에는 공통으로 사용하는 에러 문구나 placeholder를 정리해두었습니다.
- 공통으로 사용될만한 유틸리티 함수들은 따로 utils 폴더로 빼서, 중복 코드를 줄이고 최대한 같이 활용하는 방식으로 진행했습니다.
- api를 각 페이지별로 생성하지 않고 기능별로 분리했습니다.
- axios를 모듈화해서, 중복 코드를 방지하고자 했습니다.
    - 지난 프로젝트에서 axios 관련 코드를 작성할 때, 동일한 역할을 하는 인스턴스를 각각 따로 만들고 인증이 필요한 요청에는 헤더에 토큰 넣는 코드를 매번 작성하는 등 중복 코드가 많이 발생했습니다. 그래서 이번에 모듈화를 시도해봤습니다!
    - AxiosInstance를 생성하여 같이 사용
    - 인증이 필요 없는 경우에는 `defaultInstance`를 사용하고, 인증이 필요한 경우엔 `authInstance`를 사용합니다.
    - `authInstance` 응답 인터셉터에서는 401에러가 발생할 경우 refresh 토큰을 갱신하는 요청을 보냅니다.
        
        > **POST** /reissue 는 쿠키에 있는 refresh 토큰으로 리프레시가 맞는지, 유효한지, DB에 저장되어 있는지를 검사한 후 accessToken과 refreshToken을 재생성합니다.
        > 
        
         refresh 토큰이 유효하면(200) 새로 발급된 accessToken을 다시 localStorage에 저장하고 기존 요청을 재시도합니다. 
        
        유효하지 않으면 로그아웃 처리됩니다.
        
    - `authInstance` 요청 인터셉터에는 요청이 서버에 전송되기 전에 accessToken을 가져와 헤더에 인증 권한을 추가합니다. 이후 수정된 config를 반환해서 이 설정대로 요청이 이루어집니다.
    - `errorInterceptor` 를 통해 에러를 쉽게 관리할 수 있도록 했습니다. api 요청 함수에 적힐 try-catch문을 생략할 수 있어 api 함수도 간결하게 작성할 수 있었습니다.

### 개인적으로 궁금한 점

- 조현진

data가 undefined일 수 있다는 에러가 뜰 때 그걸 어떤 식으로 해결해야 좋을지 잘 모르겠습니다.

```tsx
// 제가 사용한 방식
{
  data?.businessName || (
    <p className="text-2xl">닉네임을 불러올 수 없습니다.</p>
  );
}

// 조건문 사용
if (!data) {
  return <p>조회 실패</p>;
}

// 논리연산자 사용
{
	data && ~~
}
```

저는 HostInfo.tsx, UserInfo.tsx 등의 파일에서 undefined일 수 있다는 에러가 발생하면 첫 번째 방식이나 세 번째 방식을 주로 사용했습니다. 
undefined를 제거할 수 있는 방법이 여러 개가 있는데 어떤 방식이 가장 적절한지, 
그리고 undefined를 보통 컴포넌트쪽에서 걸러내는지, 아니면 api 함수나 hook에서 먼저 걸러낸 데이터를 컴포넌트에서 사용하는지 궁금합니다!

- 이은수

API 통신 시, 서버에서 보내주는 에러 status code를 기반으로 errorInterceptor에서 분리해 처리하고 있는데, 각 에러를 어떻게 처리해야 할지 아직 감이 안 옵니다. 공통으로 에러 처리를 한 번에 하는 방식이 나을지, 아니면 각 API 요청 로직에서 에러를 개별적으로 핸들링하는 게 더 나은 방법인지 고민입니다!

devpang20 commented on src/apis/index.ts:
```diff
@@ -0,0 +1,118 @@
+const onError = (message: string): never => {
+  throw new Error(message);
+};
+
```
1. 에러처리
- 1-1 공통에러처리 에러 응답에 대해 명확하게 메세지를 전달하면 좋습니다.

```js
const errorInterceptor = (error: AxiosError) => {
  if (error.response) {
    const { status } = error.response;

    switch (status) {
      case 401: {
        onError('인증이 만료되었습니다.');
        break;
      }
      case 403: {
        onError('접근 권한이 없습니다.');
        break;
      }
      case 404: {
        onError('요청하신 정보를 찾을 수 없습니다.');
        break;
      }
      case 500: {
        onError('서버 오류가 발생했습니다.');
        break;
      }
      default: {
        onError('알 수 없는 오류가 발생했습니다.');
      }
    }
  }

  // 네트워크 에러
  if (!error.response) {
    onError('네트워크 연결을 확인해주세요.');
  }

  return Promise.reject(error);
};
```

devpang20 commented on src/apis/review.ts:
```diff
@@ -0,0 +1,67 @@
+  return response.data;
+};
+
+// 리뷰 작성
```
- 1-2 개별 에러 처리의 경우 아래와 같이 처리하면 됩니다.
```js
export const postReview = async (data: ReviewData): Promise<Review> => {
  try {
    const response = await authInstance.post('/api/v1/review', data);
    return response.data;
  } catch (error) {
    // 리뷰 작성 시 특별히 처리해야 하는 에러만 여기서 처리
    if (error.response?.status === 400) {
      throw new Error('리뷰 내용을 확인해주세요.');
    }
    // 나머지는 공통 에러 처리로 전파
    throw error;
  }
};
```

devpang20 commented on src/pages/DetailPage/components/WorkPlaceReview.tsx:
```diff
@@ -0,0 +1,54 @@
+    createdAt: '2024-11-12T14:00:00',
+  },
+];
+
```
2. data undefined  에러 처리
2-1 커스텀 hooks 도입
- src/hooks/useWorkplaceReview.ts 생성
```js
import { useQuery } from '@tanstack/react-query';
import { ReviewData } from '../types';

export const useWorkplaceReview = (workplaceId: string) => {
  const { data, isLoading, error } = useQuery({
    queryKey: ['workplaceReview', workplaceId],
    queryFn: () => getWorkplaceReview(workplaceId),
    retry: 1,
  });

  return {
    reviews: data ?? [], // undefined일 경우 빈 배열 반환
    isLoading,
    error: error as Error | null,
  };
};
```
- 2-2 hooks 도입 후 WorkPlaceReview 수정
```js
import { useWorkplaceReview } from '@/hooks/useWorkplaceReview';

const WorkPlaceReview = ({ workplaceId }: { workplaceId: string }) => {
  const { reviews, isLoading, error } = useWorkplaceReview(workplaceId);

  if (isLoading) {
    return (
      <div className='w-custom flex justify-center'>
        <p>리뷰를 불러오는 중...</p>
      </div>
    );
  }

  if (error) {
    return (
      <div className='w-custom flex justify-center'>
        <p className='text-subfont'>리뷰를 불러올 수 없습니다.</p>
      </div>
    );
  }

  if (reviews.length === 0) {
    return (
      <div className='w-custom flex justify-center'>
        <p className='text-subfont'>작성된 리뷰가 없습니다.</p>
      </div>
    );
  }

  return (
    <div className='w-custom'>
      {reviews.map((item) => (
        <div
          key={item.reviewId}
          className='border-b border-subfont py-[16px] last:border-none'
        >
          <ReviewComponent review={item} />
        </div>
      ))}
    </div>
  );
};
```

devpang20 commented on src/pages/DetailPage/components/ReviewComponent.tsx:
```diff
@@ -0,0 +1,42 @@
+    createdAt: string;
+  };
+}
+
```
- 2-3 ReviewComponent에서의 데이터 처리
```js
interface ReviewComponentProps {
  review: ReviewData;
}

const ReviewComponent = ({ review }: ReviewComponentProps) => {
  // review는 이미 타입이 보장된 상태
  const { userName, reviewRating, content, createdAt } = review;

  return (
    <div>
      <h3>{userName}</h3>
      <div>{reviewRating}</div>
      <p>{content}</p>
      <span>{createdAt}</span>
    </div>
  );
};
```

devpang20 reviewed (commented):
## 👍 잘된 점
- 규모 맞춰서 적절히 나눈 폴더 구조
- axios 인터셉터를 통한 모듈화
- 공통 컴포넌트 분리, 유틸 함수 재사용

## 🛠️ 개선할 점
1. 에러처리 강화, 공통 에러와 개별 에러를 사용할 수 있는 구조로 변경(코드 코멘트 1-n 참고) 
- 대부분의 에러는 공통 처리로 해결
- 특별한 처리가 필요한 경우만 개별 API에서 처리
- 코드 중복을 최소화하면서도 유연한 에러 처리 가능

2. 커스텀 훅을 통한 데이터 처리 로직 분리(코드 코멘트 2-n 참고)
#### 데이터 흐름이 명확해짐
- API → 커스텀 훅 → 컴포넌트
- undefined 처리가 커스텀 훅에서 이루어짐
#### 컴포넌트 책임이 분리됨
- 데이터 fetching은 커스텀 훅에서
- UI 렌더링은 컴포넌트에서
#### 에러 처리가 체계적
- 로딩/에러/빈 상태 처리가 명확함
- 사용자에게 적절한 피드백 제공
#### 타입 안정성 확보
- 컴포넌트에서는 이미 타입이 보장된 데이터 사용
- undefined 체크 불필요
이런 방식으로 구현하면 코드가 더 안정적이고 유지보수하기 쉬워질 것 같습니다.

## 추가적인 추천
- .prettierrc 설정파일과 에디터 autosave 를 통한 코드 컨벤션 자동 적용
참고
https://velog.io/@kcs0702/VSC-%EC%9E%90%EB%8F%99%EC%A0%95%EB%A0%AC-auto-save-prettier-%EC%84%A4%EC%A0%95%EB%B2%95


- API 타입 추가
```ts
export interface ApiResponse<T> {
  data: T;
  status: number;
  message: string;
}

export interface ApiError {
  code: string;
  message: string;
  status: number;
}

// API 상태 코드 상수화
export const API_STATUS = {
  SUCCESS: 200,
  UNAUTHORIZED: 401,
  FORBIDDEN: 403,
  NOT_FOUND: 404,
  SERVER_ERROR: 500,
} as const;

// 에러처리 상수키워드 이용
switch (status) {
  case API_STATUS.UNAUTHORIZED:
    return '로그인이 필요합니다.';
  case API_STATUS.FORBIDDEN:
    return '접근 권한이 없습니다.';
  case API_STATUS.NOT_FOUND:
    return '요청하신 정보를 찾을 수 없습니다.';
  case API_STATUS.SERVER_ERROR:
    return '서버 오류가 발생했습니다.';
  default:
    return '알 수 없는 오류가 발생했습니다.';
```
## 폴더구조 참고 자료
- FSD 라는 구조가 있다는 것도 참고, 현재 유명한 디렉토리 분류 기준이라 참고정도 해보면 좋음. 영상에도 나오지만 디렉토리 구조 분류에 정답은 없음. 불편할때 개선을 위해 나누는게 정답.
https://www.youtube.com/watch?v=64Fx5Y1gEOA

eunjju2 committed:
🗑️ Chore : 병합

ryeong9 committed:
✨ Feat : 룸 수정 페이지에서 룸 등록 #87

ryeong9 committed:
🗑️ Chore : 병합

eunjju2 committed:
💄 Design : 장소 모달 스타일 수정 #105

eunjju2 committed:
♻️ Refactor : 필요한 위치값 zustand로 상태 관리 개선 #105

eunjju2 committed:
♻️ Refactor : 페이지 이동해도 중심값 안바뀌도록 개선 #105

devpang20 commented on src/pages/ReservationPage/hooks/useGetPossibleTime.ts:
```diff
@@ -0,0 +1,32 @@
+import { getPossibleTime } from '@apis/workplace';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { PossibleTime } from '@typings/types';
+
+type ModifyDateType = { studyRoomId: number; checkDate: Date };
+
+export const useGetPossibleTime = (studyRoomId: number, checkDate: Date) => {
+  const { data, isLoading, isError } = useQuery<PossibleTime>({
+    queryKey: ['studyroomDetail', studyRoomId, checkDate],
+    queryFn: () => getPossibleTime(studyRoomId, checkDate),
+  });
+
+  return { data: (data ?? {}) as PossibleTime, isLoading, isError };
+};
+
+export const usePossibleTimeMutation = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation<PossibleTime, Error, ModifyDateType>({
+    mutationFn: ({ studyRoomId, checkDate }) =>
+      getPossibleTime(studyRoomId, checkDate),
+    onSuccess: (_, variables) => {
+      const { studyRoomId, checkDate } = variables;
+      queryClient.invalidateQueries({
+        queryKey: ['studyroomDetail', studyRoomId, checkDate],
+      });
+    },
+    onError: (error) => {
+      console.log(error);
+    },
+  });
+};
```
### useGetPossibleTime의 경우 데이터 조회가 목적으로 보임. 목적에 맞는 메서드 사용으로 코드 간소화
- 현재 useMutation에서 GET 요청을 사용하고 있어 역할이 맞지 않습니다.
- useMutation은 주로 데이터 생성(POST), 수정(PUT/PATCH), 삭제(DELETE)와 같은 서버의 상태를 변경하는 작업에 사용됩니다.
- 필터링 조건이 변경될 때는 useQuery의 queryKey를 변경하여 자동으로 데이터를 조회하게 하는 것이 더 효율적입니다. 그렇게 되면 invalidateQueries 역시 사용할 필요가 없습니다.

```js
import { getPossibleTime } from '@apis/workplace';
import { useQuery } from '@tanstack/react-query';
import { PossibleTime } from '@typings/types';

export const useGetPossibleTime = (studyRoomId: number, checkDate: Date) => {
  const { data, isLoading, isError } = useQuery<PossibleTime>({
    queryKey: ['studyroomDetail', studyRoomId, checkDate],
    queryFn: () => getPossibleTime(studyRoomId, checkDate),
    enabled: !!studyRoomId && !!checkDate,
  });

  return { 
    data: (data ?? {}) as PossibleTime, 
    isLoading, 
    isError 
  };
};
```
### 변경 추천로직 설명 
- useMutation, 불필요한 invalidateQueries  제거
- queryKey에 studyRoomId와 checkDate를 포함시켜서 이 값들이 변경될 때마다 자동으로 새로운 데이터를 조회합니다.

### useQuery, useMutation 사용 정리
#### useQuery 사용 시점
- 데이터 조회 (GET)
- 필터링된 데이터 조회
- 페이지네이션
- 실시간 데이터 폴링

#### useMutation 사용 시점
- 데이터 생성 (POST)
- 데이터 수정 (PUT/PATCH)
- 데이터 삭제 (DELETE)
- 서버의 상태를 변경하는 작업

eunjju2 replied:
궁금증이 명확하게 해결됐습니다, 감사합니다!! 리팩토링에 반영해보겠습니다 :)

devpang20 reviewed (commented):
tanstack-query 에 대해서 추가적으로 리뷰 달았습니다.

JOEIH committed:
✨ Feat : 알림 기능 초안

ryeong9 committed:
✨ Feat : 룸 수정 페이지 수정 및 삭제 1차 구현 #87

ryeong9 committed:
✨ Feat : 수정 페이지 기능 구현 1차

✨ Feat : 수정 페이지 기능 구현 1차

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
♻️ Refactor : 지도 데이터 받아오는 것 개선 #105

eunjju2 committed:
♻️ Refactor : 예약 가능한 시간대 조회 개선 #105

eunjju2 committed:
♻️ Refactor : 위치 기반 데이터 로딩 중일 때 처리 #105

eunjju2 committed:
✨ Feat : 상세 정보 페이지에서 룸 선택 시 페이지 이동 #105

eunjju2 committed:
✨ Feat, ♻️ Refactor : 지도 개선 및 코드 리뷰 리팩토링 적용

✨ Feat, ♻️ Refactor : 지도 개선 및 코드 리뷰 리팩토링 적용

eunjju2 committed:
♻️ Refactor : 검색페이지 현재 시간 이전 시 시간 블럭 선택 안되도록 개선 #105

JOEIH committed:
✨ Feat : 실시간 알림 연결 #104

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
🏷️ Types : 사업자의 예약자 확인에서 결제 취소 확인 위해 reservationState 추가

JOEIH committed:
✨ Feat : 결제 취소된 예약일 경우 사업자의 예약자 조회 페이지에서 결제 취소된 예약임을 표시

eunjju2 committed:
🔧 Fix : 상세 정보 룸카드 선택 시 페이지 이동 개선 #110

eunjju2 committed:
✨ Feat : 정적 경로로 임시 연결 #111

eunjju2 committed:
♻️ Refactor : 검색페이지 시간 블록 개선 #110

eunjju2 committed:
♻️ Refactor : 지도 첫 생성 시 위치값 0으로 요청되는 것 개선 #111

eunjju2 committed:
♻️ Refactor : 지도 및 검색, 상세페이지 개선

♻️ Refactor : 지도 및 검색, 상세페이지 개선

JOEIH committed:
✨ Feat : sse 타임아웃 설정 #104

JOEIH committed:
🗑️ Chore : 병합

eunjju2 committed:
🗑️ Chore : 임시 응답값으로 인한 시간 표현 주석 처리 #111

eunjju2 committed:
🔧 Fix : 이미지 캐러셀 너비 수정 #113

ryeong9 committed:
♻️ Refactor : 하단 패딩 값 안 되어 있는 페이지들 적용 #87

eunjju2 committed:
💄 Design : 상단 스크롤 메인 레이아웃에 추가 및 다른 페이지에서 삭제 #113

ryeong9 committed:
♻️ Refactor : 룸 카드의 룸 이름 길어졌을 때 말줄임 표시 #87

ryeong9 committed:
♻️ Refactor : 상세 페이지에서 리뷰 없을 때 처리

eunjju2 committed:
📝 Docs : CICD 워크플로우 작성 #114

eunjju2 committed:
📝 Docs : 워크플로우 브랜치 부분 수정 #114

eunjju2 committed:
[Docs] CICD 워크플로우 작성

[Docs] CICD 워크플로우 작성

eunjju2 committed:
📝 Docs : args 수정 #114

eunjju2 committed:
📝 Docs : 워크플로우 args 수정 

📝 Docs : 워크플로우 args 수정

eunjju2 committed:
📝 Docs : action 수정 #114

eunjju2 committed:
📝 Docs : action 수정 

📝 Docs : action 수정

eunjju2 committed:
📝 Docs : 버전에 따른 오류 발생 부분 수정 #114

eunjju2 committed:
📝 Docs : 버전에 따른 오류 발생 부분 수정

📝 Docs : 버전에 따른 오류 발생 부분 수정

JOEIH committed:
✨ Feat : 실시간 알림 수정 #104

JOEIH committed:
✨ Feat : 알림 수정 #104

JOEIH committed:
🗑️ Chore : 병합

ryeong9 committed:
♻️ Refactor : oAuth token 이름 변경

ryeong9 committed:
🗑️ Chore:  병합

JOEIH committed:
💄 Design : 실시간 알림 위치 고정 #104

eunjju2 committed:
✨ Feat : 맞춤형 추천 api 연결 #120

eunjju2 committed:
✨ Feat : 비로그인 상태 시 화면 구현 #120

eunjju2 committed:
✨ Feat : 비로그인 / 사업자 / 사용자별 화면 및 기능 분리 #120

eunjju2 committed:
💄 Design : 메인 탭 하단 라인 수정 #120

ryeong9 committed:
♻️ Refactor : 리뷰 key 변경

eunjju2 committed:
✨ Feat : 맞춤형 추천 스터디룸 지도 표시 구현 #120

ryeong9 committed:
♻️ Refactor : 상세 페이지 및 여러 피드백 반영

♻️ Refactor : 상세 페이지 및 여러 피드백 반영

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
✨ Feat : 메인페이지 맞춤형 추천 기능 구현

✨ Feat : 메인페이지 맞춤형 추천 기능 구현

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
✨ Feat : 채팅 목록 불러오기 사용자, 사업자 분리 #111

eunjju2 committed:
✨ Feat : 응답값 수정에 따른 변수명 수정 #111

eunjju2 committed:
🔧 Fix : 예약 가능한 시간대 조회 시간 수정 #113

eunjju2 committed:
🔧 Fix : 예약 가능한 시간대 조회 시간 수정 #113

JOEIH committed:
✨ Feat : 실시간 알림 전역 상태관리 구현 및 애니메이션 적용 #104

eunjju2 committed:
🔧 Fix : 검색 시 시간 블럭 업데이트 안되는 것 개선 #113

eunjju2 committed:
💄 Design : 조회된 룸이 없습니다 문구 스타일 수정 #113

eunjju2 committed:
🔧 Fix : 조회된 룸이 빈 배열일 때 처리 #113

eunjju2 committed:
💄 Design : 등록된 리뷰가 없습니다 문구 스타일 수정 #113

eunjju2 committed:
♻️ Refactor : 추천 탭 클릭 안할 때는 쿼리 비활성화로 개선 #113

eunjju2 committed:
🗑️ Chore : console.log 제거 #113

eunjju2 committed:
♻️ Refactor : 중심 위치 바뀔 때도 데이터 재요청하도록 개선 #113

eunjju2 committed:
♻️ Refactor : 지도에 보여줄 데이터 중복 코드 개선 #113

eunjju2 committed:
🔧 Fix : 비로그인 시 예약 버튼 막기 수정 #113

eunjju2 committed:
💄 Design : 결제 완료 페이지 주문 상품 스타일 수정 #113

eunjju2 committed:
🔧 Fix : import cycle 뜨는 것 개선 #113

eunjju2 committed:
🔧 Fix : 화살표 클릭 시 이동하도록 개선 #113

eunjju2 committed:
🔧 Fix : 사업자일 때 룸 선택 못하도록 수정 #113

eunjju2 committed:
🔧 Fix, ♻️ Refactor : 메인 및 검색 페이지 개선

🔧 Fix, ♻️ Refactor : 메인 및 검색 페이지 개선

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
✨ Feat : 실시간 알림 전역 상태관리 구현 및 애니메이션 적용

✨ Feat : 실시간 알림 전역 상태관리 구현 및 애니메이션 적용

JOEIH committed:
🚚 Rename : notiContainer.tsx 파일 대문자로 수정

JOEIH committed:
🚚 Rename : notiContainer.tsx 파일 대문자로 수정

🚚 Rename : notiContainer.tsx 파일 대문자로 수정

eunjju2 committed:
🗑️ Chore : NotiContainer 파일 재생성

eunjju2 committed:
🗑️ Chore : NotiContainer 파일 재생성

🗑️ Chore : NotiContainer 파일 재생성

JOEIH committed:
♻️ Refactor : 사업자 예약, 리뷰 알림 조회 api 추가 및 수정 #38

eunjju2 committed:
🔧 Fix : 상세정보 룸 카드 가격 표시 수정 #113

eunjju2 committed:
✨ Feat : 채팅 테스트 완료 #111

eunjju2 committed:
🗑️ Chore : console.log 제거

eunjju2 committed:
🔧 Fix : status 값 수정

eunjju2 committed:
✨ Feat : 채팅 기능 구현 완료

✨ Feat : 채팅 기능 구현 완료

JOEIH committed:
♻️ Refactor : 사업자 알림 타입 수정, 실시간 알림 타입 수정 #104 #38

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
♻️ Refactor : 사용자한텐 실시간 알림 안 뜨게 하기 위해 MEMBER_RESERVATION_CONFIRMED일 땐 안 보이도록 처리 #104

JOEIH committed:
✨ Feat : 사업자 알림 페이지 api 연결 및 마크업 수정 #38

ryeong9 committed:
✨ Feat : 사용자 알림 조회 api 연결 #33

eunjju2 committed:
🔧 Fix : 사업자일 때 버튼 막기 수정 #113

JOEIH committed:
♻️ Refactor : 사업자 예약 알림 타입 이름 수정 및 실시간 알림 콘솔 삭제 #38 #104

ryeong9 committed:
✨ Feat : 사용자 알림 조회 api 연결 

✨ Feat : 사용자 알림 조회 api 연결

JOEIH committed:
🗑️ Chore : 병합

eunjju2 committed:
🔧 Fix : map key값 수정 #113

eunjju2 committed:
🔧 Fix : 웹소켓 경로 수정 #129

eunjju2 committed:
🔧 Fix : 렌더링 중일 때 상태 업데이트 되는 부분 수정 #113

eunjju2 committed:
💄 Design : 채팅 읽음 처리 디자인 수정 #129

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
🔧 Fix : 채팅 기능 개선

🔧 Fix : 채팅 기능 개선

JOEIH committed:
♻️ Refactor : 실시간 알림 사용자 역할 검증 로직 추가, 예약자 확인 페이지에서 ON_HOLD는 null처리

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
✨ Feat, ♻️ Refactor  : 사업자 알림 페이지 api 연결 및 마크업 수정, 실시간 알림 수정 

✨ Feat, ♻️ Refactor  : 사업자 알림 페이지 api 연결 및 마크업 수정, 실시간 알림 수정

JOEIH committed:
🏷️ Types : 사업장 조회에 studyRoomCount 추가

eunjju2 committed:
🔧 Fix : 결제 취소 시 결제 취소 페이지로 이동

eunjju2 committed:
📝 Docs : 서버 경로 수정

eunjju2 committed:
🔧 Fix : async 삭제

eunjju2 committed:
🔧 Fix, 📝 Docs : 결제 취소 개선 및 yml 파일 수정

🔧 Fix, 📝 Docs : 결제 취소 개선 및 yml 파일 수정

eunjju2 committed:
📝 Docs : asset 및 경로 수정

eunjju2 committed:
🗑️ Chore : 불필요한 파일 삭제

eunjju2 committed:
📝 Docs : asset 및 경로 수정

📝 Docs : asset 및 경로 수정

JOEIH committed:
✨ Feat : 사업자의 사업장 조회 타입 수정,추가 및 사업장의 룸 개수 불러오는 방식 변경

eunjju2 committed:
📝 Docs : 경로 재수정

eunjju2 committed:
📝 Docs : 경로 재수정

📝 Docs : 경로 재수정

JOEIH committed:
💄 Design : 사업자 알림 페이지 디자인 수정

JOEIH committed:
💄 Design : 유저 알림 페이지 디자인 수정

JOEIH committed:
💄 Design : 헤더 높이 수정

JOEIH committed:
💄 Design : 회원가입, 로그인, 알림 헤더 및 패딩 조정

JOEIH committed:
✨ Feat : 사용자 알림 누르면 사업장 상세페이지로 이동

JOEIH committed:
💄 Design : 검색 결과 페이지 스터디룸 카드 수정

JOEIH committed:
💄 Design : 사업자, 사용자 마이페이지 버튼 위치, 최근 예약 카드 말줄임처리

JOEIH committed:
🗑️ Chore : 병합

eunjju2 committed:
♻️ Refactor : 채팅 1차 개선 #136

JOEIH committed:
✨ Feat, 💄 Design : 사업자의 사업장 조회 타입 수정,추가 및 사업장의 룸 개수 불러오는 방식 변경, 헤더 조정

✨ Feat, 💄 Design : 사업자의 사업장 조회 타입 수정,추가 및 사업장의 룸 개수 불러오는 방식 변경, 헤더 조정

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
🔧 Fix : 채팅 기존 로직으로 수정 및 console.log 제거 #136

eunjju2 committed:
💄 Design : 읽음처리 스타일 제거 #136

eunjju2 committed:
🔧 Fix : 채팅방 리스트 데이터 요청 수정 #136

eunjju2 committed:
🔧 Fix : 검색 요청 값 예외 처리 #136

eunjju2 committed:
🔧 Fix : 채팅 및 버그 개선

🔧 Fix : 채팅 및 버그 개선

JOEIH committed:
💄 Design : 헤더 hr 위치 조정

JOEIH committed:
♻️ Refactor : 사업자 알림 map uniqueKey값 부여 #38

JOEIH committed:
✨ Feat: 예약 ON_HOLD, PAYMENT_FAIL처리, paymentsucces page 수정

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
♻️ Refactor : ReservationList 콘솔 제거

JOEIH committed:
✨ Feat, 💄 Design : 예약 ON_HOLD, PAYMENT_FAIL처리, paymentsucces page 수정, 헤더 hr 위치 조정

✨ Feat, 💄 Design : 예약 ON_HOLD, PAYMENT_FAIL처리, paymentsucces page 수정, 헤더 hr 위치 조정

eunjju2 committed:
💄 Design : 주문 상품 글자 잘리는 문제 스타일 수정

eunjju2 committed:
💄 Design : 주문 상품 글자 잘리는 문제 스타일 수정

💄 Design : 주문 상품 글자 잘리는 문제 스타일 수정

JOEIH committed:
♻️ Refactor : 사용자 이메일 수정 불가능하게 처리

JOEIH committed:
♻️ Refactor : 사업자 이메일 수정 안 되게 처리

JOEIH committed:
💄 Design : 탈퇴버튼 위치 조정

JOEIH committed:
♻️ Refactor : 채팅 목록 페이지에서 목록 없으면 비어있음 문구 노출

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
♻️ Refactor : 사용자 정보 수정 후 정보 조회 페이지로 이동

JOEIH committed:
♻️ Refactor : 이메일 수정 불가능하게 처리

♻️ Refactor : 이메일 수정 불가능하게 처리

JOEIH committed:
♻️ Refactor : 알림 이미지 타입 이름 수정

JOEIH committed:
♻️ Refactor : 알림 이미지 타입 이름 수정

♻️ Refactor : 알림 이미지 타입 이름 수정

eunjju2 committed:
🔧 Fix : 비로그인일 때 버튼 오류 수정

eunjju2 committed:
🔧 Fix : 비로그인일 때 버튼 오류 수정

🔧 Fix : 비로그인일 때 버튼 오류 수정

devpang20 commented on src/apis/index.ts:
```diff
@@ -0,0 +1,124 @@
+import useAuthStore from '@store/authStore';
+import { toast } from 'react-toastify';
+
+// Default Instance
+const defaultInstance: AxiosInstance = axios.create({
+  baseURL: BASE_URL,
+  timeout: 2000,
+  headers: {
+    accept: 'application/json',
+    'Content-Type': 'application/json',
+  },
+});
```
- 설정 상수화
```js
export const AXIOS_CONFIG = {
  TIMEOUT: 2000,
  HEADERS: {
    ACCEPT: 'application/json',
    CONTENT_TYPE: 'application/json',
  },
} as const;

// src/apis/index.ts
const defaultInstance = axios.create({
  baseURL: ENV.API_BASE_URL,
  timeout: AXIOS_CONFIG.TIMEOUT,
  headers: {
    accept: AXIOS_CONFIG.HEADERS.ACCEPT,
    'Content-Type': AXIOS_CONFIG.HEADERS.CONTENT_TYPE,
  },
});
```

devpang20 commented on src/apis/index.ts:
```diff
@@ -0,0 +1,124 @@
+  withCredentials: true,
+});
+
+// response interceptor (토큰 갱신)
+authInstance.interceptors.response.use(
+  (response: AxiosResponse) => response,
+  async (error: AxiosError) => {
+    // token 갱신하기
+    if (error.response && error.response.status === 401) {
+      try {
+        const response = await authInstance.post('/reissue');
+
+        if (response.status === 200) {
+          const token = response.headers.authorization;
+          setAuthToken(token);
+
+          const originalRequest = error.config as AxiosRequestConfig;
+          if (originalRequest.headers) {
+            originalRequest.headers.Authorization = `Bearer ${token}`;
+          }
+          return await authInstance(originalRequest); // 실패했던 요청 재시도
+        }
+      } catch (refreshError) {
+        // 로그아웃 처리
+        const { storeLogout } = useAuthStore();
+        authInstance.post('/logout');
+        removeAuthToken();
+        removeRole();
+        storeLogout();
+        window.location.replace('/');
+      }
+    }
+    return Promise.reject(error);
+  },
+);
```
- 개선 제안
```js
export class AuthService {
  private static isRefreshing = false;
  private static refreshSubscribers: Array<(token: string) => void> = [];

  static async refreshToken() {
    if (this.isRefreshing) {
      return new Promise(resolve => {
        this.refreshSubscribers.push(resolve);
      });
    }

    this.isRefreshing = true;
    try {
      const response = await authInstance.post('/reissue');
      const newToken = response.headers.authorization;
      this.refreshSubscribers.forEach(cb => cb(newToken));
      this.refreshSubscribers = [];
      return newToken;
    } finally {
      this.isRefreshing = false;
    }
  }
}
```

devpang20 commented on src/components/CommonInput.tsx:
```diff
@@ -0,0 +1,36 @@
+interface InputProps {
+  label: string;
+  placeholder?: string;
+  defaultValue?: string | number;
+  value?: string;
+  onChangeFunction?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  maxLength?: number;
+  name?: string;
+}
```
- 게선제안
```js
// src/components/common/Input/types.ts
export interface BaseInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label: string;
}

// src/components/common/Input/Input.tsx
import { BaseInputProps } from './types';

const Input = ({ label, className, ...props }: BaseInputProps) => {
  return (
    <div className='input-wrapper'>
      <label className='input-label'>{label}</label>
      <input className={clsx('input-base', className)} {...props} />
    </div>
  );
};
```

devpang20 commented on src/components/NotificationProvider.tsx:
```diff
@@ -0,0 +1,28 @@
+import { getRole } from '@utils/auth';
+import NotiContainer from './NotiContainer';
+
+const NotificationProvider = () => {
+  const { isLogin } = useAuthStore();
+  const { message, connect, state } = useNotificationStore();
+  const role = getRole();
+
+  const connectSSE = useCallback(() => {
+    connect();
+  }, [connect]);
+
+  useEffect(() => {
+    if (!state) {
+      connectSSE();
+    }
+    if (isLogin && role) {
+      connectSSE();
+    }
+  }, [isLogin, connectSSE, state, role]);
```
- 개선제안
```js
// src/hooks/useNotification.ts
export const useNotification = () => {
  const { isLogin } = useAuthStore();
  const { message, connect, state } = useNotificationStore();
  const role = useRole();

  useEffect(() => {
    let cleanup = () => {};

    if (isLogin && role && !state) {
      connect();
      cleanup = () => {
        // 연결 해제 로직
      };
    }

    return cleanup;
  }, [isLogin, role, state, connect]);

  return { message, isActive: !!message };
};

// src/components/NotificationProvider.tsx
const NotificationProvider = () => {
  const { message, isActive } = useNotification();
  
  if (!isActive) return null;
  
  return <NotiContainer message={message} />;
};
```

devpang20 reviewed (approved):
여러분 너무 고생많으셨습니다. 앞으로 좋은결과 있길 바랍니다. 🙂
리뷰는 어디까지나 제안이고 필요하신부분만 선택적으로 반영하셔도 됩니다.
여러분들이 판단하여 PR을 머지하거나 닫으셔도 됩니다.

코드 코멘트 외 추가적인 제안은 아래와 같습니다.

타입 시스템 개선
```js
// src/types/api.ts
export interface ApiResponse<T> {
  data: T;
  status: number;
  message: string;
}

export interface PaginatedResponse<T> extends ApiResponse<T> {
  totalPages: number;
  currentPage: number;
  hasNext: boolean;
}

// src/apis/workplace.ts 사용 예시
export const getWorkPlace = async (
  workplaceId: number
): Promise<ApiResponse<WorkPlaceData>> => {
  const response = await defaultInstance.get<ApiResponse<WorkPlaceData>>(
    `/api/v1/workplace/info/${workplaceId}`
  );
  return response.data;
};
```
상태 관리 타입
```js
// src/store/types.ts
export interface AuthState {
  isLogin: boolean;
  role: UserRole | null;
  user: UserProfile | null;
}

export type AuthAction = 
  | { type: 'LOGIN'; payload: { role: UserRole; user: UserProfile } }
  | { type: 'LOGOUT' };
```
유틸 개선
```js
// src/utils/date.ts
export const formatDate = (date: Date): string => {
  return new Intl.DateTimeFormat('ko-KR', {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
  }).format(date);
};

// src/utils/validation.ts
export const validateEmail = (email: string): boolean => {
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
};
```
에러처리 개선
```js
// src/types/error.ts
export class AppError extends Error {
  constructor(
    public code: string,
    message: string,
    public status?: number
  ) {
    super(message);
    this.name = 'AppError';
  }
}

// src/utils/error.ts
export const handleApiError = (error: unknown): AppError => {
  if (error instanceof AppError) {
    return error;
  }
  
  if (axios.isAxiosError(error)) {
    return new AppError(
      error.response?.data?.code || 'UNKNOWN_ERROR',
      error.response?.data?.message || '알 수 없는 오류가 발생했습니다.',
      error.response?.status
    );
  }
  
  return new AppError('UNKNOWN_ERROR', '알 수 없는 오류가 발생했습니다.');
};
```

eunjju2 committed:
📝 Docs : README 작성

eunjju2 committed:
📝 Docs : README 작성

📝 Docs : README 작성

eunjju2 committed:
♻️ Refactor : index key로 사용 가능하도록 eslint 룰 개선 #144

eunjju2 committed:
♻️ Refactor : 라우터 버전 경고 개선 #144

eunjju2 committed:
♻️ Refactor : 의존성 배열 개선 #144

JOEIH committed:
♻️ Refactor : 사용자 예약 내역, 리뷰 관리, 알림 페이지 로딩중 화면 개선 #145

JOEIH committed:
♻️ Refactor : 사업자의 예약자 관리 페이지 로딩중 아이콘 추가 #145

JOEIH committed:
♻️ Refactor : 사업자의 사업장 관리 페이지 로딩중 아이콘 추가 #145

eunjju2 committed:
🚚 Rename : 컴포넌트 폴더로 이동 #144

JOEIH committed:
♻️ Refactor : 사업자의 알림 페이지 로딩중 아이콘 추가 #145

JOEIH committed:
🗑️Chore : 병합

eunjju2 committed:
♻️ Refactor : 읽음 처리 개선 중 #144

JOEIH committed:
♻️ Refactor : 로딩중 아이콘 추가

♻️ Refactor : 로딩중 아이콘 추가

eunjju2 committed:
♻️ Refactor : 코드 개선

♻️ Refactor : 코드 개선

eunjju2 committed:
♻️ Refactor : 예약 기간 3개월로 제한 #148

JOEIH committed:
💄 Design : 회원 성별 정보에 따른 폰트 색상 수정

eunjju2 committed:
🔧 Fix : 예약 가능한 시간 버그 수정 #148

JOEIH committed:
💄 Design : 리뷰 카드에 수정, 삭제 버튼 추가 #149

eunjju2 committed:
♻️ Refactor : 사업자 상세정보에 영업시간 추가 #148

JOEIH committed:
✨ Feat : 리뷰 삭제 버튼 모달창 연결 #149

eunjju2 committed:
♻️ Refactor : 인라인 스타일 코드 제거 #148

JOEIH committed:
✨ Feat : 리뷰 삭제 버튼과 api 연결 #149

JOEIH committed:
✨ Feat : 리뷰 수정 api 작성 #149

eunjju2 committed:
🗑️ Chore : console.log 제거

eunjju2 committed:
♻️ Refactor : 사용 안하는 유틸 함수 제거 #148

eunjju2 committed:
🔧 Fix : import cycle 버그 수정

eunjju2 committed:
♻️ Refactor : 예약 기능 개선

♻️ Refactor : 예약 기능 개선

JOEIH committed:
✨ Feat : 리뷰 수정 페이지 생성 및 api 연결 #149

JOEIH committed:
🗑️Chore : 병합

eunjju2 committed:
🚚 Rename : 룸 컴포넌트 폴더 이동 #151

eunjju2 committed:
🚚 Rename : 룸 컴포넌트 폴더 이동 #151

eunjju2 committed:
♻️ Refactor : 유효성 검사 공용 함수로 개선 #151

JOEIH committed:
💄 Design : 리뷰 수정 페이지에 사업장 이미지 정보 추가 #149

JOEIH committed:
🗑️ Chore : tailwind + 들어간 부분(오타) 수정

JOEIH committed:
✨ Feat : 리뷰 수정 및 삭제 기능 추가

✨ Feat : 리뷰 수정 및 삭제 기능 추가

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
🔧 Fix : 결제 페이지 스터디룸 이미지 버그 수정 #154

JOEIH committed:
💄 Design : 사용자, 사업자 알림페이지에 3달 지난 알림은 삭제처리된다는 문구 추가

eunjju2 committed:
♻️ Refactor : 결제 서비스 이용 약관 보기 추가 #154

JOEIH committed:
🗑️ Chore : notificationStore에서 사용하지 않는 activeNoti 삭제

eunjju2 committed:
🔧 Fix : 스터디룸 사진 배열이 아닐 때 예외 처리 #154

eunjju2 committed:
♻️ Refactor : 동의서 상세 보기 추가 #154

eunjju2 committed:
🔧 Fix : 사진 배열 아닐 때 처리 #154

eunjju2 committed:
♻️ Refactor : 전역 상태 새로고침해도 유지되도록 개선 #154

eunjju2 committed:
♻️ Refactor : 새로 검색 시 전역상태 초기화 #154

eunjju2 committed:
✨ Feat, 🔧 Fix, ♻️ Refactor : 결제 페이지 및 검색 결과 개선

✨ Feat, 🔧 Fix, ♻️ Refactor : 결제 페이지 및 검색 결과 개선

JOEIH committed:
♻️ Refactor : formatTime.ts 파일 안 7일 이내 판별하는 함수가 상태를 반환하도록 수정 및 알림페이지에 최근 7일 구분선 추가 #153

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
♻️ Refactor : 최근 7일 이내 알림 구분선 추가 #153

JOEIH committed:
💄 Design : 사업자 리뷰 알림에 마진 추가

JOEIH committed:
♻️ Refactor : 최근 7일 이내 알림 구분선 추가

♻️ Refactor : 최근 7일 이내 알림 구분선 추가

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
♻️ Refactor : 수정되는 이미지가 존재하는지 확인 개선 중 #151

eunjju2 committed:
♻️ Refactor : 추가되는 사진만 새로 업로드하도록 개선 #151

eunjju2 committed:
Docs 📝 : README 맞춤형 추천 화면 추가

eunjju2 committed:
Docs 📝 : README 맞춤형 추천 화면 추가

Docs 📝 : README 맞춤형 추천 화면 추가

eunjju2 committed:
🗑️ Chore : 병합

JOEIH committed:
♻️ Refactor : 예약 내역 페이지 데이터 없을 경우 로딩 시간 최적화 및 reissue 요청으로 401 응답이 오면 로그아웃 처리

JOEIH committed:
🗑️ Chore : 병합

JOEIH committed:
♻️ Refactor : 예약 내역 페이지 로딩 시간 최적화 및 reissue 요청 401 응답 로그아웃 처리

♻️ Refactor : 예약 내역 페이지 로딩 시간 최적화 및 reissue 요청 401 응답 로그아웃 처리

eunjju2 committed:
🗑️ Chore : 병합

JOEIH committed:
♻️ Refactor : 렌더링 시 isLoading 우선순위 변경 #159

JOEIH committed:
♻️ Refactor : 렌더링 시 isLoading 우선순위 변경

♻️ Refactor : 렌더링 시 isLoading 우선순위 변경

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
🔧 Fix : reissue 버그 1차 개선 #161

eunjju2 committed:
🔧 Fix : reissue 시 에러 문구 안뜨도록 개선 #161

eunjju2 committed:
🔧 Fix, ♻️ Refactor : 룸 수정 페이지 개선 및 무한 리다이렉팅 버그 개선

🔧 Fix, ♻️ Refactor : 룸 수정 페이지 개선 및 무한 리다이렉팅 버그 개선

eunjju2 committed:
♻️ Refactor : 채팅 구조에 맞게 요청 개선 #163

eunjju2 committed:
♻️ Refactor : 소켓 헤더에 필요한 값 추가 #163

eunjju2 committed:
💄 Design : 결제 버튼 스타일 수정

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
♻️ Refactor : 채팅 요청 params 수정 사항 반영 #163

eunjju2 committed:
♻️ Refactor : 헤더에 닉네임 넘겨주기 개선 #163

eunjju2 committed:
♻️ Refactor : 채팅 요청에 params 수정 사항 반영 #163

eunjju2 committed:
♻️ Refactor : 토큰 만료 시 화면 전환 개선 중 #164

eunjju2 committed:
🔧 Fix : 채팅 요청 cursor 값 잘못 넘겨주는 것 개선 #163

eunjju2 committed:
♻️ Refactor : S3 이미지 삭제 개선 #151

eunjju2 committed:
♻️ Refactor : 토큰 만료 기존 인스턴스에서 확인하도록 수정 #164

eunjju2 committed:
♻️ Refactor : cursor 값 한국 시간으로 개선 #163

eunjju2 committed:
♻️ Refactor : 메인페이지에서 토큰 만료 확인으로 개선 #164

eunjju2 committed:
♻️ Refactor : 채팅 cursor 값 수정 및 페이지네이션 1차 반영 #163

eunjju2 committed:
♻️ Refactor : S3 이미지, 토큰, 채팅 개선

♻️ Refactor : S3 이미지, 토큰, 채팅 개선

eunjju2 committed:
🔧 Fix : 소켓 disconnect 안되는 문제 해결 #163

eunjju2 committed:
♻️ Refactor : 채팅 소켓 읽음 처리를 위한 새로운 구독 추가 #163

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
♻️ Refactor : 채팅 읽음 처리 개선

♻️ Refactor : 채팅 읽음 처리 개선

eunjju2 committed:
♻️ Refactor : 채팅 URL 수정

eunjju2 committed:
♻️ Refactor : 채팅 URL 수정

♻️ Refactor : 채팅 URL 수정

eunjju2 committed:
♻️ Refactor : 맞춤형 추천 서버 안될 시 재요청 막도록 개선

eunjju2 committed:
♻️ Refactor : 맞춤형 추천 서버 안될 시 재요청 막도록 개선

♻️ Refactor : 맞춤형 추천 서버 안될 시 재요청 막도록 개선

eunjju2 committed:
🔧 Fix : 예약 시간 선택 버그 개선

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
♻️ Refactor : 요청 timeout 개선

eunjju2 committed:
🔧 Fix : 예약 시간 선택, 검색 요청 버그 개선

🔧 Fix : 예약 시간 선택, 검색 요청 버그 개선

eunjju2 committed:
🗑️ Chore : 서버 확인을 위한 log 추가

eunjju2 committed:
🗑️ Chore : 병합

eunjju2 committed:
🗑️ Chore : 서버 확인을 위한 log 추가

🗑️ Chore : 서버 확인을 위한 log 추가
